packages/vm/tests/api/EIPs/eip-3541.spec.ts

```diff
@@ -0,0 +1,144 @@
+import tape from 'tape'
+import VM from '../../../lib'
+import Common from '@ethereumjs/common'
+import { Transaction } from '@ethereumjs/tx'
+import { InterpreterStep } from '../../../lib/evm/interpreter'
+import { Address } from 'ethereumjs-util'
+
+const pkey = Buffer.from('20'.repeat(32), 'hex')
+
+tape('EIP 3541 tests', (t) => {
+  const common = new Common({ chain: 'mainnet', hardfork: 'berlin', eips: [3541] })
+  const commonNoEIP3541 = new Common({ chain: 'mainnet', hardfork: 'berlin', eips: [] })
+
+  t.test('should not deposit 0xEF code if 3541 is active', async (st) => {
+    // put 0xEF contract
+    const tx = Transaction.fromTxData({
+      data: '0x7FEF0000000000000000000000000000000000000000000000000000000000000060005260206000F3',
+      gasLimit: 1000000,
+    }).sign(pkey)
+
+    let vm = new VM({ common })
+
+    let result = await vm.runTx({ tx })
+    let created = result.createdAddress
+
+    let code = await vm.stateManager.getContractCode(created!)
+
+    st.ok(code.length === 0, 'did not deposit code')
+
+    // Test if we can put a valid contract
+
+    // put a valid contract starting with SELFDESTRUCT
+    const tx1 = Transaction.fromTxData({
+      data: '0x7FFF0000000000000000000000000000000000000000000000000000000000000060005260206000F3',
+      gasLimit: 1000000,
+      nonce: 1,
+    }).sign(pkey)
+
+    result = await vm.runTx({ tx: tx1 })
+    created = result.createdAddress
+
+    code = await vm.stateManager.getContractCode(created!)
+
+    st.ok(code.length > 0, 'did deposit code')
+
+    // check if we can deposit a contract on non-EIP3541 chains
+
+    vm = new VM({ common: commonNoEIP3541 })
+    const tx2 = Transaction.fromTxData({
+      data: '0x7FEF0000000000000000000000000000000000000000000000000000000000000060005260206000F3',
+      gasLimit: 1000000,
+      nonce: 2,
+    }).sign(pkey)
+
+    result = await vm.runTx({ tx: tx2 })
+    created = result.createdAddress
+
+    code = await vm.stateManager.getContractCode(created!)
+
+    st.ok(code.length > 0, 'did deposit code')
+
+    st.end()
+  })
+
+  t.test('should not deploy contracts starting with 0xEF using CREATE', async (st) => {
```

Comment:
These header are not matching the test cases applied with both the success and failure cases, maybe just drop the "should not" here (and other occurrences where it applies)